tiledb/sm/array/array.cc

```diff
@@ -818,20 +818,85 @@ shared_ptr<const Enumeration> Array::get_enumeration(
     return schema->get_enumeration(enumeration_name);
   }
 
-  return get_enumerations({enumeration_name}, schema)[0];
+  return get_enumerations({enumeration_name})[0];
+}
+
+std::unordered_map<std::string, std::vector<shared_ptr<const Enumeration>>>
+Array::get_enumerations_all_schemas() {
+  if (!is_open_) {
+    throw ArrayException("Unable to load enumerations; Array is not open.");
+  }
+
+  std::unordered_map<std::string, std::vector<shared_ptr<const Enumeration>>>
```

Comment:
NIT: a general comment, those complex types are easier to read if modeled as structs so that we avoid `.second` type of names that are hard to remember what they refer to. No need to change it here, just stating my preference :)

Reply:
Good call! I did not change it here but I'll keep it in mind for the future.. I have been confused by `.first` and `.second` in the past so I agree, a structure would have been better.